### DISCUSSION TITLE
Velero backup from remote cluster

### DISCUSSION THREAD

**Discussion by Akshaya-T** [vmware-tanzu/velero#8197] [Community support Q&A]
**Setup** 
- we use Argocd for deploying applications on Kubernetes.
- Terraform to create the underlying Kubernetes infra. 
- Currently the Github repo acts as the source and backup for the Kubernetes resources, so the only thing which is left is the volumes attached with the stateful resources which is what we are trying to backup and restore incase of DR scenarios.

**Cluster Info**

Central cluster - A
 - Running Velero controller
Prod Cluster - B
 - Running production workloads.

**Questions**

- Let's if I want to take periodic backups of Statefulset's PV snapshots running in Prod Cluster from the Central cluster is it possible with Velero? I could not able to find any relevant docs for the same.

**Comment by draghuram** [comment]
It is not possible to do backups and restores on a remote cluster. Velero controller only monitors its CRDs in the same cluster as it is installed in. Any particular reason you don't want to install Velero on the prod cluster? You may also want to consider other solutions (such as CloudCasa where I work) if that is feasible. Also note that even if you are ready to install Velero on the prod cluster, there are challenges to make it work using a gitops workflow. For example, you won't be able to create PVC resource using ArgoCD and then restore data using Velero. You will need to let Velero do the restore of PVC/Pod resources as well.

**Reply by Akshaya-T** [reply]
Got it, thank you for response @draghuram . 

The reason is, we want to isolate the backup controller from running within the same cluster as prod.
Our primary use case is only to take the backups of cloud provided volume snapshots (Statefulset PV's), since we will be having all the k8s resources (deployment, jobs etc ) deployed through ArgoCD ( Hence backup is not required ).

So in restore case of scenario ArgoCD takes care of creating all the stateless apps in the recovery cluster, the only gap here is to attach the volume snapshot appropriately for stateful applications ( For which we would need to find custom way of doing it ).
